src/inmanta/util/dict_path.py

```diff
@@ -271,8 +271,23 @@ def get_path_sections(self) -> Sequence["WildDictPath"]:
     def parse(cls: type[TWDP], inp: str) -> Optional[TWDP]:
         pass
 
-    def _validate_container(self, container: object) -> TypeGuard[dict[object, object]]:
-        return isinstance(container, dict)
+    def _validate_container(
+        self, container: object, set: bool = False, remove: bool = False
+    ) -> TypeGuard[dict[object, object]]:
+        """Validate that the container supports the required mapping protocol operations
+
+        TODO: fix the typeguard. Probably by providing multiple methods for validation!
+        """
+        if not hasattr(container, "__getitem__"):
+            return False
+
+        if set and not hasattr(container, "__setitem__"):
+            return False
+
+        if remove and not hasattr(container, "__delitem__"):
+            return False
```

Comment:
I'd prefer `isinstance` checks on the ABCs here, but iirc there' a separate ticket to implement this properly.